src/simplification/SimplifyTypes.ts

```diff
@@ -0,0 +1,109 @@
+import { Schema } from "models/Schema";
+
+
+/**
+ * Find the types for a simplified version of a schema
+ * 
+ * @param schema to find the simplified types for
+ */
+export default function simplifyTypes(schema: Schema | boolean) : string[] {
+  //If we find absence of data format ensure all types are returned
+  if(typeof schema === "boolean"){
+    return ["object", "string", "number", "array", "boolean", "null"];
+  }
+  let types : string[] = [];
+  
+  const addToTypes = (typesToCheck: string | string[]) => {
+    if(typeof typesToCheck === "string"){
+      if(!types.includes(typesToCheck)){
+        types.push(typesToCheck);
+      }
+    }else{
+      typesToCheck.forEach(addToTypes);
+    }
+  };
+
+  if(schema.type){
+    addToTypes(schema.type);
+  }
+
+  //If we encounter combination schemas ensure we recursively find the types
+  if(schema.allOf){
+    schema.allOf.forEach((allOfSchema) => {
+      addToTypes(simplifyTypes(allOfSchema));
+    });
+  }
+  if(schema.oneOf){
+    schema.oneOf.forEach((allOfSchema) => {
+      addToTypes(simplifyTypes(allOfSchema));
+    });
+  }
+  if(schema.anyOf){
+    schema.anyOf.forEach((allOfSchema) => {
+      addToTypes(simplifyTypes(allOfSchema));
+    });
+  }
+
+  //If we encounter combination schemas ensure we recursively find and cumulate the types
+  if(schema.then){
+    addToTypes(simplifyTypes(schema.then));
+  }
+  if(schema.else){
+    addToTypes(simplifyTypes(schema.else));
+  }
+
+  //Check enums and const keywords to infer type if type has not already been defined.
+  if(!schema.type){
+    const inferTypeFromValue = (value: any) => {
+      if(Array.isArray(value)){
+        return "array";
+      }
+      if(value === null){
+        return "null";
+      }
+      const typeOfEnum = typeof value;
+      switch(typeOfEnum){
+        //This should never happen, but just to be sure
+        case "undefined": 
+        case "function":
+        case "symbol":
+          return;
+        case "bigint": 
+          return "number";
+        default:
+          return typeOfEnum;
+      }
+    };
+    if(schema.enum){
+      types = [];
+      schema.enum.forEach((value: any) => {
+        const inferredType = inferTypeFromValue(value);
+        if(inferredType !== undefined){
+          addToTypes(inferredType);
+        }
+      });
+    }
```

Comment:
What about refs for another models etc? I mean in enum you can also add ref to another model, so whats then? Will it be always object? 😄

Reply:
As far as I know enums can never be refs. They can be objects, but never refs 😄

Reply:
We must contribute to JSON Schema 😄

Reply:
We kinda are in a sense 😄